typings/index.d.ts

```diff
@@ -783,20 +783,20 @@ declare module 'discord.js' {
 		public readonly position: number;
 		public rawPosition: number;
 		public readonly viewable: boolean;
-		public clone(options?: GuildChannelCloneOptions): Promise<GuildChannel>;
+		public clone<T extends this>(options?: GuildChannelCloneOptions): Promise<T>;
```

Comment:
Is there any reason for `public clone<T extends this>(options?: GuildChannelCloneOptions): Promise<T>;` instead of `public clone(options?: GuildChannelCloneOptions): Promise<this>;`? I use the latter in many projects of mine, and so far it works very well.